packages/data-stores/src/site/actions.ts

```diff
@@ -119,7 +125,21 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 		return success;
 	}
 
+	function* setSiteTitle( title: string, siteId: number = window._currentSiteId ) {
+		yield wpcomRequest( {
+			path: `/sites/${ encodeURIComponent( siteId ) }/settings`,
+			apiVersion: '1.4',
+			body: {
+				blogname: title,
+			},
+			method: 'POST',
+		} );
+		yield receiveSiteTitle( siteId, title );
```

Comment:
I see there is no error handling at the moment but since `receiveSiteTitle` is not executed if the request fails, the behaviour on the UI is correct: previous title is not changed.
A `try..catch` block would make sure no error gets to the console but it's up to you.